steward/src/cork/cache.rs

```diff
@@ -0,0 +1,69 @@
+use abscissa_core::{
+    tracing::{debug, error},
+    Application,
+};
+use lazy_static::lazy_static;
+use std::{collections::HashSet, sync::RwLock, time::Duration};
+use tokio::task::JoinHandle;
+
+use crate::{cork::client::CorkQueryClient, error::Error, prelude::APP};
+
+pub type ApprovedCellarsCache = RwLock<HashSet<String>>;
```

Comment:
Is there a reason this state is implemented as a `RwLock` but we don't do that for the proposal state?

Reply:
Only one thread is accessing the proposal state, while this can be accessed by any number of SP requests

Reply:
Yeah, I mean, the only accessors are here in this file and writing is only occurring in the thread, but I guess it's technically safer to implement the lock in case something down the road decided to directly interact with `APPROVED_CELLARS`.

Reply:
I guess it is only reads that are happening on other threads. I'll leave it as you say, but yeah it's probably unnecessary now that you mention it.